File: backend/app/schemas/capability.py
from typing import Any
from pydantic import BaseModel

class CapabilityResult(BaseModel):
    id: str
    label: str
    description: str
    category: str
    score: float
    decision: str
    reasons: list[str]
    signals: dict[str, Any]

class ACSASummary(BaseModel):
    total_capabilities: int
    run_count: int
    optional_count: int
    skip_count: int

class ACSAResult(BaseModel):
    dataset_id: str
    algorithm: str
    algorithm_version: str
    generated_at: str
    business_objective: dict[str, str]
    problem_type: str
    thresholds: dict[str, float]
    capabilities: list[CapabilityResult]
    summary: ACSASummary

File: frontend/src/types/capability.ts
export interface CapabilityResult {
  id: string; label: string; description: string; category: string; score: number;
  decision: "run" | "optional" | "skip"; reasons: string[]; signals: Record<string, string | number | boolean | null>;
}
export interface ACSAResult {
  dataset_id: string; algorithm: string; algorithm_version: string;
  generated_at: string; business_objective: { id: string; label: string };
  problem_type: string; thresholds: Record<string, number>;
  capabilities: CapabilityResult[];
  summary: { total_capabilities: number; run_count: number; optional_count: number; skip_count: number };
}
